docs/DeveloperGuide.md

```diff
@@ -2,14 +2,266 @@
+        getOptionByName("opt3") // -> null
+```
+
+##### `Parser`
+
+This is now just a matter of wrapping `ParsedInput` with suitable error handling and logic such that each command will
+be used to initiated a corresponding command class (e.g. `AddCommand`), while errors are handled gracefully.
+
+#### Reason for current implementation
+
+We need an intuitive, safe and declarative way to parse the user input. Alternative implementations that can only parse
+specific commands with specific options are more imperative, less readable, less maintainable and overall just a pain to
+handle. That's why the two classes are here.
+
+#### Alternative implementation
+
+No.
+
+### Review Flashcard Feature
+
+#### Current implementation
+
+The current review flashcard allows the user to review all the flashcards that are due today or before,
+it is implemented through the following steps:
+
+Step 1:
+The input of user is collected by `getUserCommand()` inside class `Ui`.
+
+Step 2:
+The input string will be converted into a `Command` object by being passed through
+`parseCommand(String userInput)` inside `Paser`.
+
+In this case, an `ReviewCommand` will be created and returned.
+
+Step 3:
+The `execute()` function of `ReviewCommand` will run, calling `getFlashCards()`
+of class `FlashcardList` to get the list of the flashcards.
+
+Then it will iterate through the `FlashcardList` and call the function `isDueBeforeToday()` of class `Flashcard` to
+check if the flashcard is due by today.
+
+If the flashcard is due by today, `reviewCurrentFlashcard(Ui display, Flashcard flashcard)` of class `ReviewCommand`
+will be called to review the card.
+
+First, the `Ui` will display the question of the current card by calling the `getQuestion()` method of
+class `Flashcard`, and ask user if user is ready to view the answer. After user enters any keyboard input, the answer of
+the current card will be shown by calling the `getAnswer()` method of class `Flashcard`, and `Ui` will ask the user if
+he/she has got the card correct. If the user inputs "y", then the current `Flashcard` is considered to be
+cleared and `updateDueDateAfterCorrectAnswer()` of `Flashcard` will be called to update its `dueDate`. Then Clanki will
+let user review the next `Flashcard`. If the user inputs "n", then the card is considered to be not cleared
+and `updateDueDateAfterIncorrectAnswer()` will be called to update its `dueDate`. Then Clanki will let user review the
+next `Flashcard`. This process will repeat until all the `Flashcards` in the `FlashcardList` are iterated.
+
+After the whole `FlashcardList` has been iterated through, a message congratulating the user that he/she has completed
+the reviewing task will be displayed.
+
+At this point, the reviewing process is completed and the program is ready to take another command.
+
+The following sequence diagram show how the review operation work:
+![ReviewFlashcard-0.png](umlDiagrams%2FReviewFlashcard-1.png)
```

Comment:
![image](https://user-images.githubusercontent.com/37980219/229274470-1aed5a82-4003-451e-9041-1d43172b678b.png)
There's a way to turn off the entities at the bottom, try to follow the notations used in our class as close as possible.